### DISCUSSION TITLE
Support Join query in filterOptions

### DISCUSSION THREAD

**Discussion by tsemachh** [payloadcms/payload#8936] [Feature Requests & Ideas]
Joins are great , yet many of the usages of joins are not possible today due to the fact that we can't use join fields with filterOptions.
filterOptions eventually use Rest API which supports join , so we need to add to filterOptions the capability to filter by join as we have in the rest API.
See:
https://discord.com/channels/967097582721572934/1300760003250094103
https://discord.com/channels/967097582721572934/1300810327574708346

**Comment by DanRibbens** [comment]
I'm making sure I understand this correctly. You want to be able to configure `filterOptions` on the `join` field so that you only surface the docs that match a certain query in the edit view join table.

I can see how that could be useful.

Is that right?

**Reply by akhrarovsaid** [reply]
Hey @DanRibbens,

Spot on - I believe this is exactly what this feature request is seeking. We have had a few users on the Discord inquire about this exact feature in recent memory for various use-cases.

**Reply by tsemachh** [reply]
@DanRibbens correct 
Imagine a researcher which gathers medias on topic x
Then author takes the data to publish article and the media in the article comes from.the author media on that topic

**Comment by DanRibbens** [comment]
https://github.com/payloadcms/payload/pull/8973

**Comment by tsemachh** [comment]
@DanRibbens I replied in https://github.com/payloadcms/payload/pull/8973#issuecomment-2453300307
You got me right besides the fact that the query is not static it's context sensitive based on the data provided in the collection field level.
aka having it in join field level will just allow us static filtering , while filterOptions provides dynamic filtering example:
```typescript
              {
                type: 'upload',
                name: 'gallery',
                relationTo: 'media',
                filterOptions: async (args: FilterOptionsProps<PublicTopic>) => {
                  'use server'
                    return { belongToTopic: { equals: args.data.topicEntity } }
                },
              },
```
While author flow is to select the topicEntity and then all media is filtered by media which has a join field belongToTopic

**Reply by tsemachh** [reply]
@DanRibbens I am returning to you on this as we have another payload app with another vertical(Finance and stocks management)
Now I am facing the same issue when I need filerOptions or dynamic where to answer the requirements
In my case I have an holding table with stocks and dates and I want to show joined per date and date comes from the main collection

**Comment by DanRibbens** [comment]
@r1tsuu has made it possible to query via the join field which is similar to the feature request here so I thought I would link it for anyone searching.

Next we can come up with the syntax and the mongodb aggregations sql joins to support @tsemachh idea https://github.com/payloadcms/payload/discussions/8936#discussioncomment-11132672.

**Reply by sviatoslav-hulko** [reply]
@DanRibbens Hey! Thanks for the update — everything works great with all columns except for id. When I try to perform any operations involving id, it seems to break. Is that expected behavior?

My use case involves using this as part of a larger query, where I only want to fetch categories that have related posts. I'm currently using this approach (and it works):

`
payload.find({
  collection: 'categories',
  where: { 'relatedPosts.title': { exists: true } },
})
`

Interestingly, this doesn't work:

`
payload.find({
  collection: 'categories',
  where: { 'relatedPosts.id': { exists: true } },
})
`

I get an error: Your "id" field references a column "posts"."id", but the table "posts" is not part of the query! Did you forget to join it?
It it a bug? And also is there a more recommended way to achieve this using joins in Payload?
